qt/python/mantidqt/mantidqt/widgets/superplot/presenter.py

```diff
@@ -360,24 +371,32 @@ def _update_list(self):
                     spectra.append(data[1])
             self._view.set_spectra_list(name, spectra)
 
-    def _update_plot(self):
+    def _update_plot(self, replot=False):  # noqa C901
         """
         Update the plot. This function overplots the memorized data with the
         currently selected workspace and spectrum index. It keeps a memory of
-        the last plot and removes it if is not part of the memorised data.
+        the last plot and removes it if is not part of the memorised data. It
+        can also replot all cuves if needed.
+
+        Args:
+            replot (bool): if True, all curves are removed and replotted
         """
         selection = self._view.get_selection()
         current_spectrum_index = self._view.get_spectrum_slider_position()
         plotted_data = self._model.get_plotted_data()
         mode = self._view.get_mode()
+        normalised = self._model.is_normalised()
 
         figure = self._canvas.figure
         axes = figure.gca()
         artists = axes.get_tracked_artists()
 
         # remove curves not in plotted_data
         for artist in artists:
-            ws, sp = axes.get_artists_workspace_and_workspace_index(artist)
+            try:
+                ws, sp = axes.get_artists_workspace_and_workspace_index(artist)
+            except KeyError:
+                continue
```

Comment:
Could you please add a comment why it is safe to ignore the `KeyError` in this case?

Reply:
This was to avoid a hard crash when more than one workspace are deleted at the same time. I did not manage to reproduce the issue when the try/catch is removed but I prefer to keep it like this. I added a comment and the commit 4d1895481644 explains it.